<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>A Dynamic Nomenclature for SARS-CoV-2 to Assist Genomic Epidemiology</title>
    <meta charset="utf-8" />
    <meta name="author" content="Adrian Zetner" />
    <link href="libs/remark-css/default.css" rel="stylesheet" />
    <link href="libs/remark-css/default-fonts.css" rel="stylesheet" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# A Dynamic Nomenclature for SARS-CoV-2 to Assist Genomic Epidemiology
## Andrew Rambaut, Edward C. Holmes, Oliver G. Pybus
### Adrian Zetner
### 2020/04/23 (updated: 2020-05-19)

---




class: inverse, center, middle

## A Dynamic Nomenclature for SARS-CoV-2 to Assist Genomic Epidemiology
### Andrew Rambaut, Edward C. Holmes, Oliver G. Pybus

#### http://virological.org/t/a-dynamic-nomenclature-for-sars-cov-2-to-assist-genomic-epidemiology/458

???

Respectively from:
University of Edinburgh
University of Sydney, Sydney, Australia.
University of Oxford, Oxford, UK.

Present a proposed solution to a problem in genomic epidemiology related to SARS-CoV-2

Image credit: [Virological](http://virological.org/t/a-dynamic-nomenclature-for-sars-cov-2-to-assist-genomic-epidemiology/458)



---
class: center, middle

## _No coherent system for naming and discussing the population diversity of SARS-CoV-2_

???

The growing number of phylogenetic lineages that comprise the population diversity of this virus, with conflicting ad hoc and informal systems of virus nomenclature in circulation. A nomenclature system is urgently required before the scientific literature is further confused.

---

## Clades

- AKA _subtypes, genotypes, groups_

- Divide pathogen phylogeny into groupings that are:
    - equally divergent
    - mutually exclusive
    - statistically well supported
    
- All genome sequences allocated to a single clade or labelled as unclassified

- Multiple hierarchical levels can exist for the same pathogen

???

Typically, pathogen genetic diversity is categorised into distinct ‘clades’, each of which corresponds to a monophyletic group on a phylogenetic tree. 

‘subtypes’, ‘genotypes’, ‘groups’, depending on taxonomic level under investigation or the established scientific literature for the infectious disease in question.

The clades usually reflect an attempt to divide pathogen phylogeny and genetic diversity into a set of groupings that are approximately equally divergent, mutually exclusive and statistically well supported.

ensure that all genome sequences are allocated to a single clade, or provisionally labelled as unclassified

Often multiple hierarchical levels of classification exist for the same pathogens, such as the terms ‘type’, ‘group’ and ‘subtype’ that are used in the field of HIV research.

---
## Hierarchical Pathogen Classification: HIV

- Two types: HIV-1 and HIV-2  

- Type 1 divided into groups M,N,O,P  

- M group divided into 9 subtypes: A-K

- Recombination between subtypes results in 'circulating recombinant forms' 

???
Two main circulating types of HIV: 1 and 2, of these 1 is the most common.
Within type 1 there are groups labelled M-P, of which M is the major group responsible for the majority of the global epidemic
The M group divided into 9 subtypes: A-K
Recombination between subtypes results in 'circulating recombinant forms' that don't fall neatly into any particular subtype.
This the _unclassified_ labelling mentioned previously. 


[hiv strains](https://www.avert.org/professionals/hiv-science/types-strains)
---

## Why Use Type/Group/Subtype Classification?

- Discussing epidemiology and transmission with stable taxonomic levels over time  
  - slowly-evolving pathogens
  - rapidly-evolving pathogens with low rates of lineage turnover

???

Such systems are useful for discussing epidemiology and transmission when the number of taxonomic labels remains roughly constant through time; this is the case for 
slowly-evolving pathogens (e.g., many bacteria) 
and for rapidly-evolving pathogens with low rates of lineage turnover (e.g. HIV and HCV). 


---

class: center, middle

## Is this approach appropriate for rapidly-evolving viruses with high rates of lineage turnover?

---
## Rapidly Evolving Viruses with High Rates of Lineage Turnover

- Rapidly evolving viruses  
  - Often RNA viruses  
  - Subject to strong evolutionary forces from host immune system  

- High rates of lineage turnover  
  - Genetic diversity currently circulating emerges out of and replaces the previous diversity
  - eg. Seasonal influenza 

???

Rapidly evolving pathogens, especially RNA viruses such as HIV-1, Influenza A virus (IAV), and Hepatitis C virus (HCV), are subject to some of the strongest evolutionary forces that have been reported in evolutionary biology. Within individual hosts, viruses undergo adaptive change at genomic sites that are targeted by the humoral (Bush et al. 1999; Frost et al. 2005), cellular (Kawashima et al. 2009), and innate immune responses. [source](https://academic.oup.com/mbe/article/35/6/1348/4983910)

In contrast, some rapidly-evolving viruses such as influenza are characterised by high rates of lineage turnover, so that the genetic diversity circulating in any particular year largely emerges out of and replaces the diversity present in the preceding few years. For human seasonal influenza, this behaviour is the result of strong natural selection among competing lineages. 

eg. Influenza where The genetic diversity circulating in any particular year largely emerges out of and replaces the diversity present in the preceding few years due to strong natural selection among competing lineages

---

class: center, middle

## Require a More Dynamic Classification System

---
## Dynamic Classification System


- Instead of Groups/Types/Subtypes uses:
  - 'subtypes'
  - 'clades'
  - 'subclades'
  
- Refers to the emergence of new potentially antigenically-distinct variants

- Common labels to viruses circulating in different regions: describe links between outbreaks by genomically similar viruses

???
In such circumstances a more dynamic classification system is used, with ‘subtypes’, ‘clades’ and ‘subclades’. Importantly, this system provides a convenient way to refer to the emergence of new potentially antigenically-distinct variants and is intimately tied into the process of selecting the component viruses for the regularly-updated influenza vaccine. A similar approach to tracking antigenic diversity may be needed to inform SARS-CoV-2 vaccine design efforts.

providing commonly-agreed labels to refer to viruses circulating in different parts of the world, thereby describing the links between outbreaks that share similar virus genomes

Future considerations of such a system will include how to describe virus lineages that vary in phenotypic or antigenic properties (although it must be stressed that at present there is no evidence of such variation among currently available SARS-CoV-2 strains).
---
## Challenges


1. Capture local and global patterns of virus genetic diversity in a timely and coherent manner  

2. Track emerging lineages as they move among countries and between populations within each country  

3. Be sufficiently robust and flexible enough to accommodate new virus diversity as it is generated  

4. Be dynamic, such that it is able to incorporate both the birth and death of viral lineages through time.


???

There are a number of key challenges in this endeavour, and to be valid accepted, a nomenclature needs to: (i) capture local and global patterns of virus genetic diversity in a timely and coherent manner, (ii) track emerging lineages as they move among countries and between populations within each country, (iii) be sufficiently robust and flexible enough to accommodate new virus diversity as it is generated, and (iv) be dynamic, such that it is able to incorporate both the birth and death of viral lineages through time.

A special challenge in the case of COVID-19 is that genome sequence data is now being generated rapidly and at high volumes, such that by the end of the pandemic we can expect tens of thousands of SARS-CoV-2 genomes to have been sequenced. Any lineage naming system must therefore be capable of handling tens to hundreds of thousands of virus genomes sampled longitudinally through time. Further, to be practical, any lineage naming system will have no more than one or two hundred active lineage labels, as any more would obfuscate rather than clarify discussion and will be difficult to conceptualise.

---

class: inverse, center, middle

## Dynamic Lineage Nomenclature

???

*We choose the term lineages (rather than ‘clades’, ‘genotypes’ or other designations) for SARS-CoV-2 as it captures the fact they are dynamic, rather than relying on a static and exclusive hierarchical structure.*

To fulfill these requirements we propose workable and practical *lineage* nomenclature for SARS-CoV-2 that arises from a set of fundamental evolutionary and phylogenetic principles. Some of these principles are, necessarily, specific to the COVID-19 pandemic, reflecting the new reality of large-scale real-time generation of virus genome sequences. 

The nomenclature system is not intended to represent every evolutionary change in SARS-CoV-2, as these will number many thousand by the end of the pandemic. Instead, the *focus is on genetic changes associated with important epidemiological and biological events.*

Fortunately, because of the early sampling and genome sequencing of COVID-19 cases in China, especially in Hubei province, it appears that the “root sequence” of SARS-CoV-2 is known. Many of the genomes from the earliest sampled cases are genetically identical and hence also likely identical to the most recent common ancestor of all sampled viruses. This occurrence is different to previous viruses and epidemics and provides some advantages for the development of a rational and scalable classification scheme. Specifically, setting the “reference sequence” to be the “root sequence” forms a natural starting point, as direct comparisons in the number and position of mutations can be made with respect to the root sequence.

---
## Focus on Certain Lineages 

- Those that contribute most to:   
  - global transmission 
  - genetic diversity

- eg. B.1 in Italy 

???
However, a central component of a useful nomenclature system is that it focuses on those virus lineages that contribute most to global transmission and genetic diversity. Hence, rather than naming every new possible lineage, classification focuses on those that have exhibited onward spread in the population, particularly those that have seeded an epidemic in a new location. For example, the large epidemic in Lombardy, northern Italy, which is thought to have begun in early February (Zehender et al. 2020) 18, has since been disseminated to other locations in northern Europe and elsewhere.

eg. Those that have been found to have seeded an epidemic in a new location such as the large epidemic in Northern Italy that then spread beyond to other regions

---
## Focus on Certain Lineages 

- Those that are currently circulating:
  - mark lineages active or inactive
  - track those contributing most to the epidemic

- eg. B.4 in Iran 

???
Further, *because SARS-CoV-2 genomes are being generated continuously and at a similar pace to changes in virus transmission and epidemic control* efforts, we expect to see a *continual process of lineage generation and extinction through time*. Rather than maintaining a cumulative list of all lineages that have existed since the start of the pandemic, it is more prudent to mark lineages as active or inactive, a designation reflecting our current understanding of whether they are actively transmitting in the population or not. Although this will allow us to track those lineages that are contributing most to the epidemic, and so reduce the number of names in use, it is important to keep open the possibility that new lineages will appear through the generation of virus genomes from unrepresented locations or from cases with travel history from such locations. 

eg. the epidemic in Iran, designated B.4 in our system, has been identified via returning travellers to other countries (Eden et al. 2020) 11, 

---
## Lineage Designations

- Start with a letter representing the original two lineages of SARS-CoV-2

- Lineage A: share two nucleotides (positions 8782 in ORF1ab and 28144 in ORF8) with the closest known bat virus (RaTG13)
  - eg. Wuhan/WH04/2020 (EPI_ISL_406801), sampled on 2020-01-05 

- Lineage B: Differ in the nulceotides present at these positions 
  - eg. Wuhan-Hu-1 (Genbank MN908947) *the reference sequence for most analyses*

???

We propose that lineage labels begin with a letter. At the root of the phylogeny of SARS-CoV-2 are two lineages that we simply denote as lineages A and B. The earliest lineage A viruses, such as Wuhan/WH04/2020 (EPI_ISL_406801), sampled on 2020-01-05, share two nucleotides (positions 8782 in ORF1ab and 28144 in ORF8) with the closest known bat virus (RaTG13). Different nucleotides are present at those sites in viruses assigned to lineage B, of which Wuhan-Hu-1 (Genbank accession MN908947) sampled on 2019-12-26 is an early representative. Hence, although viruses from lineage B were sequenced and published first, it is likely (based on current data) that lineage A viruses form the root of the SARS-CoV-2 pandemic phylogeny. At the time of writing, viruses from both lineages A and B are still circulating in many countries around the world, reflecting the exportation of viruses from Hubei to other regions of China and elsewhere before the strict travel restrictions and quarantine measures were imposed there.

---
## Lineage Designations

- Based on ML tree rooted between lineages A and B
- Descendents of each are assigned numerical values (eg. A.1) as follows:
  1. Phylogentic evidence of emergence into a geographically distinct population  
    a. one or more shared nucleotide differences from the ancestral lineage  
    b. at least 5 genomes with &gt;95% of the genome sequenced  
    c. genomes within the lineage exhibit at least one shared nucleotide change among them
    d. a bootstrap value &gt;70% for the lineage defining node  
  2. Lineages identified in step 1 can act as ancestors to new populations resulting in a new lineage designation (e.g A.1.1)  
  3. Step 2 can be repeated a maximum of 4 times creating sublevels of that lineage (e.g. A.1.1.1.1) 
    - Afterwards a new descendent lineages are assigned a letter and the process continues: the descendent of A.1.1.1.1 would become C.1  
  4. Any genomes that are not assigned to these lineages are given the designation A.U or B.U for “unassigned”

???

1. Each descendent lineage should show phylogenetic evidence of emergence from an ancestral lineage into another geographically-distinct population, implying substantial onward transmission in that population. In the case of a rapidly-expanding global lineage the recipient “population” may in fact comprise multiple countries. In the case of large and populous countries it may represent a new region or province. To show phylogenetic evidence a new lineage must meet all of the following criteria: (a) it exhibits one or more shared nucleotide differences from the ancestral lineage, (b) it comprises at least 5 genomes with &gt;95% of the genome sequenced, © genomes within the lineage exhibit at least one shared nucleotide change among them, and (d) a bootstrap value &gt;70% for the lineage defining node. 
Criterion © helps to focus attention only on lineages with evidence of on-going transmission.

2. The lineages identified in step (I) can themselves act as ancestors for virus lineages that then emerge in other geographic areas or at later times, provided they satisfy criteria a-d above. This results in a new lineage designation (e.g A.1.1).

3. The iterative procedure in step II can proceed for a maximum of 4 sublevels (e.g. A.1.1.1.1) after which new descendent lineages are given a letter (in English alphabetical sequence from C - so A.1.1.1.1 would become C.1) and the process continues. The rationale for this is that the system is intended only for tracking currently circulating lineages, hence we do not try to capture the entire history of a lineage in its label (that complete history can be obtained by reference to a phylogeny).

4. Any genomes that are not assigned to these lineages are given the designation A.U or B.U in which ‘U’ denotes “unassigned”). In lineage A and B the unassigned viruses include all those from Hubei province, China, as well as any viruses from many national and international travellers from this area.
Similar to the 'circulating recombinant forms' of HIV


---
class: inverse, center, middle

## Applying the Nomenclature

---
## Identified Lineages from A


- 5 Lineages derived from A (A.1-A.5)

- 2 Additional lineages derived from A.1 (Australia, Canada, Iceland, USA)  
  - A.1.1: Iceland  
  - A.1.2: Canada, USA  

???

Using this scheme we identified 5 lineages derived from lineage A (denoted lineage A.1 – A.5) and 9 lineages derived from lineage B (B.1 – B.9). We also identified two additional lineages – A.1.1 and A.1.2 - descended from lineage A.1, 16 further lineages descended from B.1 (B.1.1 to B.1.11), as well sub-lineages B.2.1, B.2.2, B.3.1, B.4.1, and B.4.2. Importantly, we are not yet able to further subdivide the other lineages, even though some of these contain very large numbers of genomes. This is because many parts of the world are experiencing numerous imported cases followed by exponential growth in local transmission. We provide descriptions of these initial lineages, including their geographical locations and time span of sampling, in Table 1. We have also tried to be liberal with the criteria where, for example, the bootstrap value is below 70% but there is strong prior evidence that the lineage exists and is important. In particular the Italian epidemic comprises two large lineages in our scheme – B.1 and B.2 – demonstrated by genomes from Italy and also from large numbers of travellers from these regions falling in both lineages.


---

## Identified Lineages from B

- 9 Lineages derived from B (B.1-B.9)

- 16 Further sublineages from B.1 (Large Italian Outbreak)

- 5 More sublineages from B.2 (Italy), B.3 (UK), B.4 (Iran)
  
  
???

Using this scheme we identified 5 lineages derived from lineage A (denoted lineage A.1 – A.5) and 9 lineages derived from lineage B (B.1 – B.9). We also identified two additional lineages – A.1.1 and A.1.2 - descended from lineage A.1, 16 further lineages descended from B.1 (B.1.1 to B.1.11), as well sub-lineages B.2.1, B.2.2, B.3.1, B.4.1, and B.4.2. Importantly, we are not yet able to further subdivide the other lineages, even though some of these contain very large numbers of genomes. This is because many parts of the world are experiencing numerous imported cases followed by exponential growth in local transmission. We provide descriptions of these initial lineages, including their geographical locations and time span of sampling, in Table 1. We have also tried to be liberal with the criteria where, for example, the bootstrap value is below 70% but there is strong prior evidence that the lineage exists and is important. In particular the Italian epidemic comprises two large lineages in our scheme – B.1 and B.2 – demonstrated by genomes from Italy and also from large numbers of travellers from these regions falling in both lineages.

---
## Further considerations:

- Regularly assess status of circulating lineages: only visulaize currently circulating lineages (seq'd within one month) to reduce noise 

- Recombination can be used to denote start of a new lineage


???

A unique and important aspect of our proposed nomenclature is that the status of the currently circulating lineages be assessed at regular intervals, with decisions made about identifying new lineages and flagging those we believe are likely not to be circulating any more due to a considerable time interval - for example, greater than one month - since the sequencing of any members. When visualising the epidemic we suggest that these lineages should be no longer labelled to reduce both the number of names in circulation and visual noise, and to focus on the current epidemiological situation.

While we regard this proposed nomenclature as practical and robust, it is important to recognise that phylogenetic inference carries statistical uncertainty and much of the available genome data is noisy, with incomplete genome coverage and errors arising from the amplification and sequencing processes. We have proposed a genome coverage threshold for proposing new lineages (see above), and we further suggest that sequences are not ascribed a lineage designation unless the genome coverage of that sequence exceeds 70% of the coding region. As noted above, when SARS-CoV-2 genetic diversity is low during the early pandemic period, there will be a direct association between lineage assignation and the presence of particular sets of mutations (with respect to the root sequence). This should help with the development of algorithmic genome labelling approaches. This task will become more complex, but still tractable, as SARS-CoV-2 genetic diversity accumulates, increasing the chance of homoplasies and reverse mutations. Classification algorithms based on patterns of mutations may be practical if they are frequently cross-checked and validated against phylogenetic reconstructions.

---
background-image: url(https://github.com/hCoV-2019/pangolin/raw/master/docs/logo.png)
background-position: 50% 50%
background-size: 500px


.footnote[Assigning Lineages with Pangolin [https://github.com/hCoV-2019/pangolin](https://github.com/hCoV-2019/pangolin)]

???

The following external software is run as part of pangolin:

iqtree

mafft

snakemake

???

[IQTREE Docs](http://www.iqtree.org/doc/Frequently-Asked-Questions)
The ultrafast bootstrap (UFBoot) feature (-bb option) was published in (Minh et al., 2013). One of the main conclusions is, that UFBoot support values are more unbiased: 95% support correspond roughly to a probability of 95% that a clade is true. So this has a different meaning than the normal bootstrap supports (where you start to believe in the clade if it has &gt;80% BS support). For UFBoot, you should only start to believe in a clade if its support is &gt;= 95%. Thus, the interpretations are different and you should not compare BS% with UFBoot% directly.

Moreover, it is recommended to also perform the SH-aLRT test (Guindon et al., 2010) by adding -alrt 1000 into the IQ-TREE command line. Each branch will then be assigned with SH-aLRT and UFBoot supports. One would typically start to rely on the clade if its SH-aLRT &gt;= 80% and UFboot &gt;= 95%.

---

## Assigning Lineages with Pangolin

- `conda activate /Drives/P/conda_envs/pangolin`


```r
pangolin: Phylogenetic Assignment of Named Global Outbreak LINeages

positional arguments:
  query

optional arguments:
  -h, --help                      show this help message and exit
  -o OUTDIR, --outdir OUTDIR      Output directory
  -d DATA, --data DATA            Optional data directory to overwrite the default
                                  package data. Minimally containing a fasta alignment
                                  and guide tree
  -n, --dry-run                   Go through the motions but dont actually run
  -f, --force                     Overwrite all output
  -t THREADS, --threads THREADS   Number of threads
```

???

No Notes


---

## Assigning Lineages with Pangolin

- Output is a csv file with taxon name and lineage assigned, one line corresponding to each sequence in the fasta file provided

- Also included are bootstrap support values for the lineage assignment.  
  - SH-aLRT test result  
  - Ultrafast Bootstrap



---

&lt;table&gt;
 &lt;thead&gt;
  &lt;tr&gt;
   &lt;th style="text-align:left;"&gt; taxon &lt;/th&gt;
   &lt;th style="text-align:left;"&gt; lineage &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; SH-alrt &lt;/th&gt;
   &lt;th style="text-align:right;"&gt; UFbootstrap &lt;/th&gt;
  &lt;/tr&gt;
 &lt;/thead&gt;
&lt;tbody&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL5672/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 80 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3575/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 95 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3380/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 95 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL6980/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 85 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 69 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL7512/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 94 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL0654/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 71 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3695/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 67 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3536/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 94 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL1095/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 92 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3692/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 80 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL7513/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 95 &lt;/td&gt;
  &lt;/tr&gt;
  &lt;tr&gt;
   &lt;td style="text-align:left;"&gt; hCoV-19/Canada/ON_PHL3350/2020 &lt;/td&gt;
   &lt;td style="text-align:left;"&gt; B.1 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 100 &lt;/td&gt;
   &lt;td style="text-align:right;"&gt; 69 &lt;/td&gt;
  &lt;/tr&gt;
&lt;/tbody&gt;
&lt;/table&gt;

- UFboostrap values are considered to be more unbiased (Minh et al., 2013) and correspond roughly to probability that the clade is true  
- SH-alrt test values above 80% are considered reliable and that the branch being studied provides a significant likelihood gain  
- Lineage assignment starts to become reliable at SH-aLRT &gt;= 80% and UFbootstrap &gt;= 95%

???

[IQTREE Docs](http://www.iqtree.org/doc/Frequently-Asked-Questions)
The ultrafast bootstrap (UFBoot) feature (-bb option) was published in (Minh et al., 2013). One of the main conclusions is, that UFBoot support values are more unbiased: 95% support correspond roughly to a probability of 95% that a clade is true. So this has a different meaning than the normal bootstrap supports (where you start to believe in the clade if it has &gt;80% BS support). For UFBoot, you should only start to believe in a clade if its support is &gt;= 95%. Thus, the interpretations are different and you should not compare BS% with UFBoot% directly.

Moreover, it is recommended to also perform the SH-aLRT test (Guindon et al., 2010) by adding -alrt 1000 into the IQ-TREE command line. Each branch will then be assigned with SH-aLRT and UFBoot supports. One would typically start to rely on the clade if its SH-aLRT &gt;= 80% and UFboot &gt;= 95%.

---
class: inverse, center, middle

fin
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
